lib/errors/aggregate-error.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const BaseError = require('./base-error');
+
+let level = 0;
+
+/**
+ * Validation Error. Thrown when the sequelize validation has failed. The error contains an `errors` property,
+ * which is an array with 1 or more ValidationErrorItems, one for each validation that failed.
+ *
+ * @param {string} message Error message
+ * @param {Array} [errors] Array of ValidationErrorItem objects describing the validation errors
+ *
+ * @property errors {ValidationErrorItems[]}
+ */
+class AggregateError extends BaseError {
```

Comment:
We should probably make a breaking change here and not extend from error AND array and instead pass a readonly array of errors into the constructor and make them publicly available to the consumer eg.

This ensures only lazy initialisation which is usually better for performance, it also makes the API easier to model in TS.

```js
class AggregateError extends BaseError {
  constructor(message, errors) { super(message); this.errors = errors; } ... }
```

Reply:
Hi, I didn't read the discussion in-depth, but since you're talking about `AggregateError`, what if we just use [`sindresorhus/aggregate-error`](https://github.com/sindresorhus/aggregate-error)?

Yes, I know, I am a fan of his repositories...

Reply:
That comes with at least additional 2 extra dependencies and cannot inherit from the sequelize error base.